### PR TITLE
Refactored other PHPUnit method calls to work with namespaced PHPUnit 6

### PR DESCRIPTION
**Pull request by peterrehm**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Continued work to make Symfony PHPUnit 6 compatible.

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 1]
```diff
@@ -41,6 +41,10 @@ public function testSpecifications($file, $expected, $yaml, $comment, $deprecate
                 if (E_USER_DEPRECATED !== $type) {
                     restore_error_handler();
 
+                    if (class_exists('PHPUnit\Util\ErrorHandler')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php]: should check underscore version

File: src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php [hunk 2]
```diff
@@ -84,7 +84,12 @@ public function testGetThrowParameterNotFoundException($parameterKey, $exception
             'fiz' => array('bar' => array('boo' => 12)),
         ));
 
-        $this->setExpectedException('Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException', $exceptionMessage);
+        if (method_exists($this, 'expectException')) {
+            $this->expectException('Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php]: can you switch to `::class` as much as possible in this PR please?

Comment (peterrehm) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php]: On all occurences of expectException/setExpectedException?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php]: yes

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php]: but only those changed in this PR, not the other one that should be kept asis to reduce future merge conflicts

**Comment by nicolas-grekas** [comment]
Thank you @peterrehm.

**Comment by peterrehm** [comment]
In case needed in the future:

````
sed -E "s/setExpectedException\('([^']*)'/setExpectedException(\1:class/" *
$this->setExpectedException(Symfony\Component\Form\Exception\InvalidArgumentException:class, 'The child with the name "foo" does not exist.');
````

Also I took care of all occurences but the MockObjects.

$ grep PHPUnit_ * -r | grep -v MockObject

Or with the pattern `PHPUnit_(?!Framework_MockObject)`

**Comment by xabbuh** [comment]
@peterrehm Isn't that missing the second colon before `class`?

**Comment by peterrehm** [comment]
It isnt, it was missing the second colon 😃

**Pull request closed without merging**